app/src/main/java/com/example/smartshopper/services/RTDBService.java

```diff
@@ -1,12 +1,24 @@
 package com.example.smartshopper.services;
 
+import android.util.Log;
+import androidx.annotation.NonNull;
+
+import com.example.smartshopper.LocalStorage;
 import com.example.smartshopper.common.Constants;
 import com.example.smartshopper.models.Comment;
 import com.example.smartshopper.models.Deal;
 import com.example.smartshopper.models.User;
+import com.example.smartshopper.responseInterfaces.BoolInterface;
+import com.example.smartshopper.responseInterfaces.ObjectInterface;
+import com.example.smartshopper.responseInterfaces.UserInterface;
+import com.google.firebase.database.DataSnapshot;
+import com.google.firebase.database.DatabaseError;
 import com.google.firebase.database.FirebaseDatabase;
 import com.google.firebase.database.Query;
-import com.google.firebase.database.ServerValue;
```

Comment:
this seems to have been dropped from the merge. this needs to be in place for upvoting functions.

Reply:
resolved